### ISSUE TITLE
Kaffe very very little patch

### ISSUE THREAD

**Issue by ezequielgarcia** [kaffe/kaffe#1]
Hi,
I have downloaded Kaffe as part of a larger research-like project. I am testing it right now on my amd64.
Just wanted to tell you:
- first: great stuff!
- second: I had to do some minor changes to compile it. I am not a gnu hacker but it seems that CPStringBuilder must be used insted of StringBuilder, at least for classpath 0.98.

Anyway, I am attaching a patch if it is of your interest. I am not very familiar either with git, but I think "git diff > " does the job, right?
Hope the patch is any good.

diff --git a/kaffe/kaffevm/constants.c b/kaffe/kaffevm/constants.c
index 7a74533..cf6e9f1 100644
--- a/kaffe/kaffevm/constants.c
+++ b/kaffe/kaffevm/constants.c
@@ -183,7 +183,7 @@ readConstantPool(Hjava_lang_Class\* this, classFile\* fp, errorInfo *einfo)
            readu4(&d4b, fp);

 #if SIZEOF_VOID_P == 8
-           if(check_if_need_toswitch_words_in_jdouble())
- ```
        if(check_if_need_to_switch_words_in_jdouble())
        pool[i] = WORDS_TO_LONG(d4b, d4);
      else
        pool[i] = WORDS_TO_LONG(d4, d4b);
  ```
  
  diff --git a/libraries/javalib/vmspecific/java/lang/reflect/Constructor.java b/libraries/javalib/vmspecific/java/lang/reflect/Constructor.java
  index 6bc0522..5339497 100644
  --- a/libraries/javalib/vmspecific/java/lang/reflect/Constructor.java
  +++ b/libraries/javalib/vmspecific/java/lang/reflect/Constructor.java
  @@ -39,6 +39,7 @@ exception statement from your version. */
  package java.lang.reflect;
  
  import gnu.java.lang.ClassHelper;
  +import gnu.java.lang.CPStringBuilder;
  
  import gnu.java.lang.reflect.MethodSignatureParser;

@@ -231,7 +232,7 @@ public final class Constructor
   public String toString()
   {
     // 128 is a reasonable buffer initial size for constructor
-    StringBuilder sb = new StringBuilder(128);
-    CPStringBuilder sb = new CPStringBuilder(128);
   Modifier.toString(getModifiers(), sb).append(' ');
   sb.append(getDeclaringClass().getName()).append('(');
   Class[] c = getParameterTypes();
  @@ -253,7 +254,7 @@ public final class Constructor
  }
  
  /\* FIXME[GENERICS]: Add X extends GenericDeclaration and TypeVariable<X> */
-  static void addTypeParameters(StringBuilder sb, TypeVariable[] typeArgs)
-  static void addTypeParameters(CPStringBuilder sb, TypeVariable[] typeArgs)
  {
   if (typeArgs.length == 0)
     return;
  @@ -269,7 +270,7 @@ public final class Constructor
  
  public String toGenericString()
  {
-    StringBuilder sb = new StringBuilder(128);
-    CPStringBuilder sb = new CPStringBuilder(128);
   Modifier.toString(getModifiers(), sb).append(' ');
   addTypeParameters(sb, getTypeParameters());
   sb.append(getDeclaringClass().getName()).append('(');
  diff --git a/libraries/javalib/vmspecific/java/lang/reflect/Field.java b/libraries/javalib/vmspecific/java/lang/reflect/Field.java
  index 3d0f377..710e3b0 100644
  --- a/libraries/javalib/vmspecific/java/lang/reflect/Field.java
  +++ b/libraries/javalib/vmspecific/java/lang/reflect/Field.java
  @@ -39,6 +39,7 @@ exception statement from your version. */
  package java.lang.reflect;
  
  import gnu.java.lang.ClassHelper;
  +import gnu.java.lang.CPStringBuilder;
  
  import gnu.java.lang.reflect.FieldSignatureParser;

@@ -210,7 +211,7 @@ extends AccessibleObject implements Member
   public String toString()
   {
     // 64 is a reasonable buffer initial size for field
-    StringBuilder sb = new StringBuilder(64);
-    CPStringBuilder sb = new CPStringBuilder(64);
   Modifier.toString(getModifiers(), sb).append(' ');
   sb.append(ClassHelper.getUserName(getType())).append(' ');
   sb.append(getDeclaringClass().getName()).append('.');
  @@ -220,7 +221,7 @@ extends AccessibleObject implements Member
  
  public String toGenericString()
  {
-    StringBuilder sb = new StringBuilder(64);
-    CPStringBuilder sb = new CPStringBuilder(64);
   Modifier.toString(getModifiers(), sb).append(' ');
   sb.append(getGenericType()).append(' ');
   sb.append(getDeclaringClass().getName()).append('.');
  diff --git a/libraries/javalib/vmspecific/java/lang/reflect/Method.java b/libraries/javalib/vmspecific/java/lang/reflect/Method.java
  index fe436da..c1c15be 100644
  --- a/libraries/javalib/vmspecific/java/lang/reflect/Method.java
  +++ b/libraries/javalib/vmspecific/java/lang/reflect/Method.java
  @@ -39,6 +39,7 @@ exception statement from your version. */
  package java.lang.reflect;
  
  import gnu.java.lang.ClassHelper;
  +import gnu.java.lang.CPStringBuilder;
  
  import gnu.java.lang.reflect.MethodSignatureParser;

@@ -282,7 +283,7 @@ extends AccessibleObject implements Member, GenericDeclaration
   public String toString()
   {
     // 128 is a reasonable buffer initial size for constructor
-    StringBuilder sb = new StringBuilder(128);
-    CPStringBuilder sb = new CPStringBuilder(128);
   Modifier.toString(getModifiers(), sb).append(' ');
   sb.append(ClassHelper.getUserName(getReturnType())).append(' ');
   sb.append(getDeclaringClass().getName()).append('.');
  @@ -308,7 +309,7 @@ extends AccessibleObject implements Member, GenericDeclaration
  public String toGenericString()
  {
   // 128 is a reasonable buffer initial size for constructor
-    StringBuilder sb = new StringBuilder(128);
-    CPStringBuilder sb = new CPStringBuilder(128);
   Modifier.toString(getModifiers(), sb).append(' ');
   Constructor.addTypeParameters(sb, getTypeParameters());
   sb.append(getGenericReturnType()).append(' ');

Thanks for all your work.

Ezequiel.

**Comment by penberg** [comment]
Why not fork Kaffe on Github and send a pull request? The changes look good to me.

**Comment by penberg** [comment]
Fixed by commit cc3c8ebe4da8359c9692da94796376ec440ad3dc ("Fix compilation errors in 'java.lang.reflect' classes"). Please send pull requests or patches generated by 'git format-patch' in the future.

**Comment by ezequielgarcia** [comment]
OK I'll do that.

Thanks.
